src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutMissingTagCheck.java

```diff
@@ -0,0 +1,193 @@
+package org.openstreetmap.atlas.checks.validation.linear.edges;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.AreaTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.ServiceTag;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Angle;
+import org.openstreetmap.atlas.utilities.tuples.Tuple;
+
+/**
+ * Flags edges that are closed shaped and have minimum of two intersections with navigable roads.
+ * {@link RoundaboutMissingTagCheck#MINIMUM_INTERSECTION} connections. See
+ * https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout for more information about
+ * roundabouts
+ *
+ * @author vladlemberg
+ */
+
+public class RoundaboutMissingTagCheck extends BaseCheck<Long>
+{
+    // Instructions
+    public static final String MISSING_JUNCTION_TAG_INSTRUCTION = "This edge might be a roundabout";
+    // Minimum intersection with Navigable Roads
+    private static final int MINIMUM_INTERSECTION = 2;
+    private static final int MODULUS = 10;
+    private static final int FIRST_EDGE_SECTION = 1;
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(MISSING_JUNCTION_TAG_INSTRUCTION);
```

Comment:
nit: asList could be replaced with Collections.singletonList as it has just one item in it.